src/main/java/Parser.java

```diff
@@ -0,0 +1,114 @@
+import Exceptions.*;
+import Exceptions.InvalidCommandException; // warning is shown in javadocs if not imported
+import Exceptions.InvalidTaskException; // warning is shown in javadocs if not imported
+import Exceptions.NoDescriptionException; // warning is shown in javadocs if not imported
+import Tasks.Deadline;
+import Tasks.Event;
+import Tasks.Task;
+import Tasks.Todo;
+
+/**
+ * Parser that helps with parsing user inputs as Strings to various
+ * Objects and types.
+ */
+public class Parser {
+
+    /**
+     * Parses the given line from file as a String and
+     * returns a Task object.
+     *
+     * @param lineFromFile The given line from the file.
+     * @return task The Task object created by the line after parsing.
+     * @throws IllegalArgumentException If the line is invalid.
+     */
+    public static Task parseTaskFromFileString(String lineFromFile) throws DukeException {
+        Task task;
+        String[] lineParts = lineFromFile.split(" \\| ");
+        String taskType = lineParts[0];
+        boolean isDone = lineParts[1].equals("X");
+        String description = lineParts[2];
+        switch (taskType) {
```

Comment:
I think that you should not indent your case statements. According to the coding standards, case statements should be on the same level of indentation as the switch statement.